### PR TITLE
Update CHANGELOG format and parsing

### PR DESCRIPTION
**Pull request by JakeQuilty**

### CODE REVIEW

**Comment by JakeQuilty** [comment]
This is failing from a different commit than the one I just made.
Commit: https://github.com/cyberark/secrets-provider-for-k8s/commit/778b95ef9f1d79b8bf26314d3e85711371b9c512

File: Jenkinsfile [hunk 1]
```diff
@@ -43,6 +43,14 @@ pipeline {
       }
     }
 
+    stage('Validate') {
```
Comment (doodlesbykumbi) [inline Jenkinsfile]: This should probably be one of the first stages to run. Right now it seems like the build is failing for other reasons and so you're unable to tell if validation is succeeding or not.

File: CHANGELOG.md [hunk 2]
```diff
@@ -9,25 +9,25 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
...
-    - More messages to increase UX and supportability
+- Logs
+  - Logging in different log levels (info, debug, warn)
+  - Capability to Enable debug logs via the env
+  - More messages to increase UX and supportability
   - An end-to-end integration test
-    
+
 ### Changed
   - Escape secrets with backslashes before patching in k8s
```
Comment (doodlesbykumbi) [inline CHANGELOG.md]: nit: prune leading space

File: Jenkinsfile (hunk 1 above)
Comment (doodlesbykumbi) [inline Jenkinsfile]: This should probably be one of the first stages to run. Right now it seems like the build is failing for other reasons and so you're unable to tell if validation is succeeding or not.

Comment (diverdane) [inline Jenkinsfile]: Looks like the `Validate` stage still needs to be moved up in the sequence, maybe even before `Build client Docker image`?

File: CHANGELOG.md [hunk 3]
```diff
@@ -9,25 +9,25 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
...
-    
\ No newline at end of file
+
+[Unreleased]: https://github.com/cyberark/secrets-provider-for-k8s/compare/v0.4.0...HEAD
+[0.4.0]: https://github.com/cyberark/secrets-provider-for-k8s/compare/v0.3.0...v0.4.0
+[0.3.0]: https://github.com/cyberark/secrets-provider-for-k8s/compare/v0.2.0...v0.3.0
```
Comment (diverdane) [inline CHANGELOG.md]: Can add a link for `0.2.0`, and put square brackets around `0.2.0` in line 19:
```
[0.2.0]: https://github.com/cyberark/secrets-provider-for-k8s/releases/tag/v0.2.0
```

**Pull request merged**